docs/mutations.md

```diff
@@ -37,133 +38,233 @@ any outstanding TODOs before opening issues.
-of different random mutational processes on top of a single simulated topology,
-or if we have obtained the tree sequence from another program and wish to
-overlay neutral mutations on this tree sequence.
+One of the most powerful features of `msprime` is the flexibility of specifying
+the model under which mutations are simulated. Select from among
+the many predefined {ref}`models <sec_mutations_models>`, such as the 
+{ref}`infinite alleles model <sec_mutations_mutation_infinite_alleles>`, or
+{ref}`define your own <sec_mutations_matrix_mutation_models_details>`.
 
 
 (sec_mutations_rate)=
 
 ## Specifying rates
 
-```{eval-rst}
-.. todo:: Documentation on the ``rate`` parameter with examples.
+The `rate` parameter of {func}`.sim_mutations` determines the rate of mutations per unit
+of sequence length per generation. A rate must be specified for mutations to be generated.
 
+```{code-cell}
+ts = msprime.sim_ancestry(2, random_seed=1)
+mts = msprime.sim_mutations(ts, rate=1, random_seed=1)
+mts.num_mutations
 ```
 
+Running {func}`.sim_mutations` on this simulated tree of length 1 with `rate=1`
+produces 7 mutations.
+
+Specifying higher rates lead to proportionately more mutations:
+
+```{code-cell}
+ts = msprime.sim_ancestry(2, random_seed=1)
+mts = msprime.sim_mutations(ts, rate=5, random_seed=1)
+mts.num_mutations
+```
+
+It's also possible to provide a {class}`.RateMap` which specifies variable
+mutation rates over different stretches of the sequence. 
+
+```{code-cell}
+ts = msprime.sim_ancestry(10, sequence_length=7, random_seed=1)
+pos = [0, 2, 5, 7]
+rate = [1, 0, 0.5]
+ratemap = msprime.RateMap(pos, rate)
+mts = msprime.sim_mutations(ts, rate=ratemap, random_seed=1)
+mts.tables.sites.position
+```
+
+In this example, the mutation rate between positions 0 and 2 is 1,
+between 2 and 5 is 0, and between 5 and 10 is 0.5.
+
+Be sure that the final position of the {class}`.RateMap` is the same as the
+`sequence_length` of the tree sequence you're adding mutations to!
+
 (sec_mutations_randomness)=
 
 ## Controlling randomness
 
-```{eval-rst}
-.. todo:: Documentation on the ``seed` parameter with examples. Can
-    link to the randomness section in ancstry.rst. Alternatively we
-    don't bother with this section and just link to the ancestry.rst
-    one from the ``seed`` parameter to mutate.
+As described in {ref}`sec_ancestry_random_seed`, ``msprime`` uses a
+[psuedorandom number generator](<https://en.wikipedia.org/wiki/Pseudorandom_number_generator>)
+from the [GNU Scientific Library](<https://www.gnu.org/software/gsl/doc/html/rng.html>) for
+random number generation. Passing an integer to the ``random_seed`` parameter makes the
+defines a trajectory for a call of {func}`.sim_mutations`, making output deterministic.
+
+```{code-cell}
+mts_1 = msprime.sim_mutations(ts, rate=1, random_seed=7)
+mts_2 = msprime.sim_mutations(ts, rate=1, random_seed=8)
+print(np.array_equal(mts_1.tables.mutations.time, mts_2.tables.mutations.time))
```

Comment:
You need to think a bit about this example to get it. What if we showed the site and mutation tables for a small example, so people can see they are the same? And then in the next cell show them for a different seed?